### ISSUE TITLE
Anchor height not available; scan required

### ISSUE THREAD

**Issue by HonzaR** [zcash/librustzcash#904]
Before we start syncing blocks with the latest FFI changes [cd391db](https://github.com/zcash/zcash-android-wallet-sdk/pull/1116/commits/cd391dbded90e229dc9d0758badf9be427f8c06c) in the [Fast-spendability](https://github.com/zcash/zcash-android-wallet-sdk/pull/1116) PR of Android SDK, we approach to refresh  the transparent balance and other things like this:
```
        launch(CoroutineExceptionHandler(::onCriticalError)) {
            refreshUtxos(Account.DEFAULT)
            refreshTransparentBalance()
            refreshTransactions()
        }
```
But after that commit, I see this error:
```
cash.z.ecc.android.sdk.demoapp.mainnet main SdkSynchronizer.onCriticalError(): Critical error occurred
java.lang.RuntimeException: Anchor height not available; scan required.
at cash.z.ecc.android.sdk.internal.jni.RustBackend.getVerifiedTransparentBalance(Native Method)
at cash.z.ecc.android.sdk.internal.jni.RustBackend.access$getVerifiedTransparentBalance(RustBackend.kt:19)
at cash.z.ecc.android.sdk.internal.jni.RustBackend$Companion.getVerifiedTransparentBalance(Unknown Source:0)
at cash.z.ecc.android.sdk.internal.jni.RustBackend$Companion.access$getVerifiedTransparentBalance(RustBackend.kt:369)
at cash.z.ecc.android.sdk.internal.jni.RustBackend$getVerifiedTransparentBalance$2.invokeSuspend(RustBackend.kt:206)
at cash.z.ecc.android.sdk.internal.jni.RustBackend$getVerifiedTransparentBalance$2.invoke(Unknown Source:8)
at cash.z.ecc.android.sdk.internal.jni.RustBackend$getVerifiedTransparentBalance$2.invoke(Unknown Source:4)
at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78)
at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:167)
at kotlinx.coroutines.BuildersKt.withContext(Unknown Source:1)
at cash.z.ecc.android.sdk.internal.jni.RustBackend.getVerifiedTransparentBalance(RustBackend.kt:205)
at cash.z.ecc.android.sdk.internal.TypesafeBackendImpl$getDownloadedUtxoBalance$verified$1.invokeSuspend(TypesafeBackendImpl.kt:131)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
at java.lang.Thread.run(Thread.java:1012)
```

The error is handled on the SDK side, but `refreshTransactions` won't happen after that caught exception from the rust layer.

- Testnet/Mainnet doesn't matter (both impacted)
- New/older wallet doesn't matter (both impacted)

Is that an expected behavior now?

**Comment by nuttycom** [comment]
@HonzaR Can you verify whether updateChainTip is being called prior to this interaction? Basically everything depends upon `updateChainTip` at this point. That being said, I think it would be reasonable for the implementation of the various `get*Balance` calls to simply return zeros if no chain data is available, instead of throwing an error. #900 will also modify the balance APIs in a way that will resolve this problem, but in general if you can ensure that `updateChainTip` is being called prior to retrieving balances that will ensure that a variety of errors similar to this one are not reachable.

**Comment by nuttycom** [comment]
I want to revisit this problem once #907 is merged and integrated into the wallet SDKs, because that integration work will require that the order of operations here is handled correctly.

**Comment by HonzaR** [comment]
> @HonzaR Can you verify whether updateChainTip is being called prior to this interaction?

@nuttycom No, the `updateChainTip` operation is being called after this initial balance and transactions poll. That is a new SbS requirement. The error is thrown by the rust layer. I'll file a new issue on making the `updateChainTip` the first operation in our algorithm. Then the https://github.com/zcash/zcash-android-wallet-sdk/issues/1180 issue will change our approach to calling this operation and also https://github.com/zcash/librustzcash/issues/900 will change when we trigger the balances and transactions refreshing as well.

**Comment by nuttycom** [comment]
Confirmed fixed by @HonzaR
